fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/ContentObject.java

```diff
@@ -26,16 +29,50 @@
  *
  */
 public abstract class ContentObject extends RepositoryObject {
+	
+	// the object containing MODS and (possibly also) source metadata
+	private FileObject fileObj;
+
+	private BinaryObject mods;
 
 	protected ContentObject(PID pid, Repository repository, RepositoryObjectDataLoader dataLoader) {
 		super(pid, repository, dataLoader);
 	}
 
-	public BinaryObject addDescription(InputStream contentStream) {
-		return null;
+	public FileObject addDescription(InputStream modsStream) {
+		fileObj = createFileObject();
+		
+		BinaryObject mods = fileObj.addOriginalFile(modsStream, null, "text/xml", null);
```

Comment:
Add a pcdm:hasRelatedObject relationship from this contentObject to the description fileObject.  See image_1 in the diagram for an example.  The purpose of the hasRelatedObject is to indicate ownership from the content object to everything in the description object, while the iana:describedby is there to point out specifically which part of the description object we are using to describe the contentObject.